### PR TITLE
🏃 Reenable building and testing CAPD

### LINKED ISSUE #1837: Add targets to build/test/verify test/infra/docker
**User Story**

As a [developer] I would like to run `make` commands from the cluster-api directory to execute build/test/verify commands in test/infrastructure/docker.

**Anything else you would like to add:**

Also hook up to CI.

/kind feature

### PR DESCRIPTION
**Pull request by ncdc**
**What this PR does / why we need it**:
Reenable building and testing CAPD

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1837

### CODE REVIEW

**Comment by vincepri** [comment]
Is there any significant time implication in having CAPD building and testing as part of the CAPI jobs?

**Comment by ncdc** [comment]
The times for the jobs are all still quite reasonable. It's possible the test job may have increased from 3-4 minutes to 6-7 minutes. But it's more important that we keep capd green. And if we need to split the capd parts into separate prow jobs, we can. I don't think we're at that point yet, though.

**Comment by vincepri** [comment]
LGTM, @ncdc did you want to squash commits?

**Comment by ncdc** [comment]
Do you want me to squash?

**Comment by chuckha** [comment]
looks good to me. A squash would be good here. Generally as a maintainer I ask people to reduce their commits down to one since one commit will usually fix one issue. If I don't always ask people to reduce commits down to one then we end up in a situation where sometimes we ask people to reduce their commits down to one because we generally don't like what their commits look like. This can be pretty subjective so it's easier to have a rule to always ask to reduce to one commit per issue closed.

**Comment by ncdc** [comment]
Squashed!

File: scripts/ci-test.sh [hunk 1]
```diff
@@ -29,3 +29,11 @@ source ./scripts/fetch_ext_bins.sh
 fetch_tools
 setup_envs
 make generate test
+# TODO(ncdc): create a separate verify prow job
+make verify
```
Comment (vincepri) [inline scripts/ci-test.sh]: We have a verify prow job that runs `make verify`, so this shouldn't be needed

File: scripts/ci-test.sh [hunk 2]
```diff
@@ -29,3 +29,11 @@ source ./scripts/fetch_ext_bins.sh
...
+
+# Docker provider
+cd test/infrastructure/docker
```
Comment (vincepri) [inline scripts/ci-test.sh]: Can we print out some new lines and a bold message of what we're testing before we run these commands?

File: scripts/ci-test.sh (hunk 1 above)
Comment (ncdc) [reply scripts/ci-test.sh]: We actually don't. We have pr-verify, but not a postsubmit or periodic version.

File: scripts/ci-test.sh (hunk 2 above)
Comment (ncdc) [reply scripts/ci-test.sh]: Like "hey we're testing capd now?"

Comment (vincepri) [reply scripts/ci-test.sh]: Yeah, just a log line that we can quickly search for

File: scripts/ci-test.sh (hunk 1 above)
Comment (vincepri) [reply scripts/ci-test.sh]: Not sure if I missed it, but the comment didn't mention postsubmit/periodic

Comment (ncdc) [reply scripts/ci-test.sh]: I can reword

Comment (ncdc) [reply scripts/ci-test.sh]: Actually, I think you're right. CAPA doesn't have a postsubmit verify either. I'll remove it.

File: test/infrastructure/docker/Makefile [hunk 3]
```diff
@@ -229,11 +244,8 @@ clean-release: ## Remove the release folder
 .PHONY: verify
 verify:
 	./hack/verify-boilerplate.sh
-	./hack/verify-doctoc.sh
-	./hack/verify-shellcheck.sh
 	$(MAKE) verify-modules
 	$(MAKE) verify-gen
-	$(MAKE) verify-modules
```
Comment (vincepri) [inline test/infrastructure/docker/Makefile]: Shouldn't we keep verify-modules given that we're removing the check in the root Makefile?

Comment (ncdc) [reply test/infrastructure/docker/Makefile]: Why do we need to verify twice?

Comment (ncdc) [reply test/infrastructure/docker/Makefile]: From the top Makefile diff:
```
 ## --------------------------------------
 ## Docker
@@ -357,11 +356,10 @@ verify:
        ./hack/verify-shellcheck.sh
        $(MAKE) verify-modules
        $(MAKE) verify-gen
-       $(MAKE) verify-modules
```

Comment (vincepri) [reply test/infrastructure/docker/Makefile]: The other line diff from the top Makefile removes `test/infrastructure/docker/go.sum` check

Comment (ncdc) [reply test/infrastructure/docker/Makefile]: Will fix

Comment (ncdc) [reply test/infrastructure/docker/Makefile]: Oh wait, yeah, I stand by my original statement. This diff is removing a second call to `verify-modules`.

Comment (vincepri) [reply test/infrastructure/docker/Makefile]: ah ok, makes sense now :)

**Pull request merged**